californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -365,6 +385,15 @@ public UdpMatcher(final NetworkConfig config) {
 			}
 			// ignore response
 			return null;
+		} else if (isResponseRelatedToMulticastRequest(exchange)) {
```

Comment:
You're right. What about:
````
} else if (isResponseRelatedToRequest(exchange, responseContext)) {
  if (exchange.getCurrentRequest().isMulticast()) {
    // keep exchange open because we might receive additional responses
    LOGGER.log(Level.FINER, "Received response for multicast request: {0}", response);
  } else {
    Exchange prev = deduplicator.findPrevious(idByMID, exchange);
    if (prev != null) {
      LOGGER.log(Level.FINER, "Received duplicate response for open exchange: {0}", response);
      response.setDuplicate(true);
    } else {
      idByMID = new KeyMID(exchange.getCurrentRequest().getMID());
      exchangesByMID.remove(idByMID);
      LOGGER.log(Level.FINE, "Closed open request [{0}]", idByMID);
    }
  }
  ...
````
BTW I have merged #98 and #102 so it would be nice if you could re-base your PR on master in order to pick up the changes ...